File: Sources/AIMeterApp/Views/GeminiInstallationGuide.swift
import AIMeterCore
import Foundation
import SwiftUI

enum GeminiInstallationGuide {
    static let url = URL(string: "https://antigravity.google/docs/cli/install/")!
    static let installCommand = "curl -fsSL https://antigravity.google/cli/install.sh | bash"

    static func instructions(for state: ServiceAccountConnectionState, localizer: AppLocalizer = AppLocalizer(language: .english)) -> [String] {
        let signIn = localizer.text("Run agy and complete Google sign-in.")
        let finish = localizer.text("Return to AI Token Meter and choose Retry.")
        switch state {
        case .connected:
            return []
        case .signInRequired:
            return [signIn, finish]
        case .notInstalled, .checking, .unavailable:
            return [installCommand, signIn, finish]
        }
    }
}

struct GeminiInstallationHelp: View {
    @Environment(\.locale) private var locale
    let state: ServiceAccountConnectionState

    var body: some View {
        let instructions = GeminiInstallationGuide.instructions(for: state, localizer: AppLocalizer(locale: locale))
        if !instructions.isEmpty {
            VStack(alignment: .leading, spacing: 5) {
                ForEach(Array(instructions.enumerated()), id: \.offset) { _, instruction in
                    if instruction == GeminiInstallationGuide.installCommand {
                        Text(instruction)
                            .monospaced()
                            .textSelection(.enabled)
                    } else {
                        Text(instruction)
                    }
                }
            }
            .aiMeterFont(.caption)
            .foregroundStyle(.secondary)
        }
    }
}

File: windows/src/settings/geminiInstallationGuide.ts
import type { UsageSnapshot } from "../state/usage"

export const GEMINI_INSTALLATION_GUIDE_LABEL = "Antigravity CLI installation guide"
export const GEMINI_INSTALL_COMMAND = "irm https://antigravity.google/cli/install.ps1 | iex"

export type GeminiSetupState = "connected" | "signInRequired" | "notInstalled" | "checking" | "unavailable"

export function geminiInstallationInstructions(state: GeminiSetupState): string[] {
  const signIn = "Run agy and complete Google sign-in."
  const finish = "Return to AI Token Meter and choose Check Status."
  if (state === "connected") return []
  if (state === "signInRequired") return [signIn, finish]
  return [GEMINI_INSTALL_COMMAND, signIn, finish]
}

export function geminiSetupStateForSnapshot(snapshot: Pick<UsageSnapshot, "status" | "statusMessage">): GeminiSetupState {
  if (snapshot.status === "fresh") return "connected"
  if (snapshot.status === "cached") {
    const message = snapshot.statusMessage?.toLowerCase() ?? ""
    return message.includes("sign in required") || message.includes("authentication required") ? "signInRequired" : "connected"
  }
  if (snapshot.status === "authenticationRequired") return "signInRequired"
  if (snapshot.status === "notInstalled") return "notInstalled"
  if (snapshot.status === "refreshing") return "checking"
  return "unavailable"
}
